### PR TITLE
Update acknowledgements and prepare `v0.7.0` release

### LINKED ISSUE #165: Release `v0.7`
# Checklist for `v0.7` release features

 - [x] Vectorisation (#116)
 - [x] Add space charge (#142)
 - [ ] ~~Add nice interface for space charge~~ (delayed to later release) (#160)
 - [x] Add custom transfer map element (#100)
 - [x] Fix issue with default length (#116, #143, #164)
 - [x] Add uniformly distributed beam initialisation (#146)
 - [x] Fix NumPy 2 compatibility (#177)

Let's discuss what features we want to have ready before releasing `v0.7`.

### PR DESCRIPTION
**Pull request by jank324**
## Description

Updates the acknowledgements, adding new authors and funding notices.

## Motivation and Context

- [x] I have raised an issue to propose this change (required for new features and bug fixes)

The current acknowledgements do not reflect the work of new contributors who have joined the project. Addresses #267. Closes #165.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (_required for a bug fix or a new feature_).
- [ ] I have updated the documentation accordingly.
  
  
- [ ] I have reformatted the code and checked that formatting passes (**required**).
- [ ] I have have fixed all issues found by `flake8` (**required**).
- [ ] I have ensured that all `pytest` tests pass (**required**).
- [ ] I have run `pytest` on a machine with a CUDA GPU and made sure all tests pass (**required**).
- [ ] I have checked that the documentation builds (**required**).

Note: We are using a maximum length of 88 characters per line.

### CODE REVIEW

**Comment by jank324** [comment]
@cr-xu, @AnEichler, @Hespe, @RemiLehe, @jp-ga: Please let me know if you have any funding strings or similar that you would like to have mentioned here. If you have nice proposals (e.g. examples from other repositories) for formatting the acknowledgements section, they are also welcome. For now I would list the authors, maybe participating institutions, and provide the funding strings.

**Comment by jp-ga** [comment]
This is the funding string from my side: 
This work was supported by the U.S. National Science Foundation under Award PHY-1549132, the Center for Bright Beams.

**Comment by Hespe** [comment]
The current XFEL project is called
> XFEL R&D Project RP-513: Learning Based Methods

**Comment by jank324** [comment]
I've put together what I think is a nice Acknowledgements section now. I know the logos don't render. The absolute paths are on purpose to fix image rendering on PyPI and should fix themselves automatically as soon as this PR is merged onto `master`.

Let me know if anything is missing. If I don't get any comments, I will merge this as is on Friday before lunch Hamburg time.

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,13 +69,24 @@ This is a major release with significant upgrades under the hood of Cheetah. Des
...
+- Refactor definitions of physical constants (see #189) (@Hespe)
 - Fix the quadrupole strength units in the quadrupole docstring (see #202) (@ansantam)
-- Add CI runs for macOS (arm64) and Windows (see #226) (@cr-xu, @jank324, @hespe)
+- Add CI runs for macOS (arm64) and Windows (see #226) (@cr-xu, @jank324, @Hespe)
 - Clean up CI pipelines (see #243, #244) (@jank324)
 - Fix logo display in README (see #252) (@jank324)
-- Made `Beam` an abstract class (see #284) (@hespe)
+- Made `Beam` an abstract class (see #284) (@Hespe)
 - Releases are now automatically archived on Zenodo and given a DOI (@jank324)
+- The Acknowledgements section in the README has been updated to reflect new contributors (see #304) (@jank324)
```
Comment (AnEichler) [inline CHANGELOG.md:79]:
```suggestion
- The Acknowledgements section in the README has been updated to reflect new contributors (see #304) (@jank324, @AnEichler )
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -57,13 +69,24 @@ This is a major release with significant upgrades under the hood of Cheetah. Des
...
+
+### 🌟 First Time Contributors
+
+- Grégoire Charleux (@greglenerd)
+- Remi Lehe (@RemiLehe)
+- Axel Huebl (@ax3l)
+- Juan Pablo Gonzalez-Aguilera (@jp-ga)
+- Andrea Santamaria Garcia (@ansantam)
+- Ryan Roussel (@roussel-ryan)
+- Christian Hespe (@Hespe)
```
Comment (AnEichler) [inline CHANGELOG.md:89]:
```suggestion
- Christian Hespe (@Hespe)
- Annika Eichler (@AnEichler)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -57,13 +69,25 @@ This is a major release with significant upgrades under the hood of Cheetah. Des
...
+- Refactor definitions of physical constants (see #189) (@Hespe)
 - Fix the quadrupole strength units in the quadrupole docstring (see #202) (@ansantam)
-- Add CI runs for macOS (arm64) and Windows (see #226) (@cr-xu, @jank324, @hespe)
+- Add CI runs for macOS (arm64) and Windows (see #226) (@cr-xu, @jank324, @Hespe)
 - Clean up CI pipelines (see #243, #244) (@jank324)
 - Fix logo display in README (see #252) (@jank324)
-- Made `Beam` an abstract class (see #284) (@hespe)
+- Made `Beam` an abstract class (see #284) (@Hespe)
 - Releases are now automatically archived on Zenodo and given a DOI (@jank324)
+- The Acknowledgements section in the README has been updated to reflect new contributors (see #304) (@jank324, @AnEichler )
```
Comment (jank324) [inline CHANGELOG.md:79]:
```suggestion
- The Acknowledgements section in the README has been updated to reflect new contributors (see #304) (@jank324, @AnEichler)
```

**Pull request merged**